.github/workflows/ci.yml

```diff
@@ -15,81 +18,58 @@ jobs:
+
+      - run: cargo fmt --all -- --check
+      - run: cargo clippy --all-features --workspace --tests --examples -- -D clippy::all
 
   lint-python:
     name: Lint Python (ruff, mypy)
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - uses: actions/setup-python@v3
-      - name: Set up venv
-        run: make ci.setup_venv
-      - name: Build and run lint
-        run: |
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+      - run: make ci.setup_venv
+
+      - run: |
           maturin develop
           make lint.python
 
-  test:
-    name: Test
+  doctest:
+    name: Documentation (and Tests)
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
+    env:
+      RUSTDOCFLAGS: -Dwarnings
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Install Rust
-        uses: dtolnay/rust-toolchain@nightly
-        with:
-          toolchain: nightly
-      - name: Run Rust tests
-        run: |
-          cargo test
+      - uses: actions/checkout@v4
 
-      - uses: actions/setup-python@v3
-      - name: Set up venv
-        run: make ci.setup_venv
-      - name: Run Python tests
-        run: |
-          maturin develop
-          pytest
-
-  # This job runs tests, generates coverage data, and generates JUnit test
-  # results in a single test invocation and then uploads it all to Codecov.
-  # However, it doesn't print test results to stdout. If Codecov's failed test
-  # reporting is solid and we never need to see the results in the CI logs, we
-  # can delete the "normal" test step and just use this.
-  test-for-codecov:
-    name: Test (Upload to Codecov)
+      - run: rustup toolchain install nightly --profile minimal --no-self-update
+
+      - run: cargo test --workspace --all-features --doc
+      - run: cargo doc --workspace --all-features --document-private-items --no-deps
+
+  test:
+    name: Test
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Install Rust
-        uses: dtolnay/rust-toolchain@nightly
-        with:
-          toolchain: nightly
-      - name: Install `cargo llvm-cov`
-        uses: taiki-e/install-action@cargo-llvm-cov
-      - name: Run tests
-        run: |
-          cargo install cargo2junit
-          cargo llvm-cov --lcov --output-path core.lcov -- -Z unstable-options --format json --report-time | cargo2junit > core-test-results.xml
+      - uses: actions/checkout@v4
+
+      - run: rustup toolchain install nightly --profile minimal --no-self-update
+      - uses: taiki-e/install-action@cargo-llvm-cov
+      - uses: taiki-e/install-action@nextest
+
+      # FIXME(swatinem): We should pass `--all-targets` to also compile and tests benchmarks
+      # Though currently `divan` does not support all CLI arguments as used by `nextest`,
```

Comment:
you moved us off `divan` in another PR, is this still relevant?

Reply:
You are right, I forgot to fix that in the followup PR, will do :-)